### ISSUE TITLE
--installdir option for valid directory leads to "does not exist" error

### ISSUE THREAD

**Issue by Daedalus359** [haskell/cabal#6526]
I am trying to update my cabal-install to the latest version. I am attempting to use the v2-install command for this purpose. The default command complains that installdir is not defined:
```
$ ~/.cabal/bin/cabal v2-install cabal-install
Resolving dependencies...
Up to date
cabal: installdir is not defined. Set it in your cabal config file or use
--installdir=<path>
```

However, when I attempt to specify an install directory as recommended, this is the result:
```
$ ~/.cabal/bin/cabal v2-install cabal-install --installdir=~/.cabal/bin
Resolving dependencies...
Up to date
~/.cabal/bin: createDirectory: does not exist (No such file or directory)
```

I also tried writing out the path as /home/kevin/.cabal/bin. Same result.

As you can see from the particular cabal binary I use to run these commands, ~/.cabal/bin does exist on my computer. Does anyone know why this is happening, or am I using the command incorrectly? It may be up to date already, but this still seems like the wrong behavior.

**System information**
 - Ubuntu 18.04
 - Cabal version 3.0.0.0
 - GHC 8.0.2

**Comment by phadej** [comment]
Compare,

```sh
% ls '~/.vim'
ls: cannot access '~/.vim': No such file or directory
% ls ~/.vim  
autoload  bundle  ftdetect  snippets  syntax
```

Don't put the equals sign between `--installdir` and the path, so the shell will expand the `~` for you, or use  `--installdir=$HOME/.cabal/bin`.

**Comment by Daedalus359** [comment]
Thanks for that - it worked. On the [relevant docs page](https://cabal.readthedocs.io/en/latest/nix-local-build.html#cabal-v2-install), an example usage is given:

```
 cabal v2-install exe:cabal --install-method=copy --installdir=~/bin
```

Why would something like that work? It seems to do the same thing that was wrong in my case (equals sign and ~ in the path).

**Comment by phadej** [comment]
That's a mistake in a docs, thanks for pointing out. I'll fix that as I work on #6519
